### ISSUE TITLE
put everything in namespace, create a single-file minimized build

### ISSUE THREAD

**Issue by kzahel** [kzahel/web-server-chrome#2]

**Comment by matt-cook** [comment]
Any plans to roll this into a build? Would love it integrate it into a project (https://github.com/zebradog/kiosk) but it's currently a bit complex. Alternatively, simple install instructions would be helpful.

**Comment by kzahel** [comment]
Hi Matt,
How does it fit into your kiosk app? This issue?
https://github.com/zebradog/kiosk/issues/9
It would likely only take a few hours for me to complete this work, but I
am unlikely to complete this in a timely manner without a bounty of some
sort :-)
p.s. You can get email notifications for reviews to your app here:
https://www.chromebeat.com/app/kiosk/afhcomalholahplbjhnmahkoekoijban

**Comment by matt-cook** [comment]
Yes, exactly. We currently have a basic server built in for serving the admin page (https://github.com/zebradog/kiosk/blob/master/src/js/main.js#L53) but we're looking to expand it to also serve files from a user-selected local directory across restarts. 

Thanks for the heads-up on Chromebeat. Handy.

**Comment by kzahel** [comment]
Everything is in the WSC namespace now...
So almost there :-)

**Comment by kzahel** [comment]
I added https://github.com/kzahel/web-server-chrome/blob/master/minimize.sh
It concatenates all the necessary files for using as a library. It does not currently minimize.

The "library" version does not support background operation. But for the kiosk, I imagine this is not a problem

**Comment by matt-cook** [comment]
Works great here! Are you referring to [`background` permissions](https://developer.chrome.com/apps/declare_permissions#background) apps? 

We're considering add this permission to our app to work around this issue: https://github.com/zebradog/kiosk/issues/27

**Comment by kzahel** [comment]
I think you don't need to worry about it, since kiosk apps are always in the "foreground"

the web server app can continue to run even if all its windows are closed. It detects an onSuspend event and cancels it which triggers an onSuspendCanceled.

Regarding #27, if chrome isn't running (eg on windows), not much you can do, unless chrome is actually still running. (they ask you on windows if you want to keep chrome apps running in the background). But yes I could see it applying in this case, but it seems like a not-too-important feature.

**Comment by matt-cook** [comment]
Thanks for the detail. Agreed, not a hugely important use-case. I'll be re-writing to call 'chrome.runtime.restart()` to do a proper restart on ChromeOS kiosk mode, and simply close and re-open the browser window on other systems.

**Comment by kzahel** [comment]
run minimize.sh to create a single file which can be included in other projects.
